src/protocol.cpp

```diff
@@ -134,6 +134,11 @@ ServiceFlags GetDesirableServiceFlags(ServiceFlags services) {
     return ServiceFlags(NODE_NETWORK | NODE_WITNESS);
 }
 
+ServiceFlags StatelessServicesFlags()
+{
+    return ServiceFlags(NODE_NETWORK | NODE_WITNESS);
```

Comment:
7c20e65b2880a23d753b3f05fd28994353b3049d

While you decouple the two, do you mean to make them independent from each other? It seems to me they will always be roughly the same. I agree with the goal of the PR, just not sure about this approach.

Reply:
> While you decouple the two, do you mean to make them independent from each other? It seems to me they will always be roughly the same. I agree with the goal of the PR, just not sure about this approach.

Yeah. The goal is to make them independent from each other so the peer manager runtime logic doesn't affect the hardcoded seeds. They are built from another source and depend on the way we externally collect them. These peers are stored in db with the stateless service flags, and would be an error to store them with a different one (they might not support the service).